feat(auth): use new firebase login methods

kadikraman opened the pull request:
### Description

As of [react-native-firebase v3.2.3](https://github.com/invertase/react-native-firebase/releases/tag/v3.2.3), `signInWithCustomToken` is deprecated in favour of `signInAndRetrieveDataWithCustomToken`.

![screen shot 2018-06-06 at 13 50 59](https://user-images.githubusercontent.com/6534400/41040097-59e6f718-6993-11e8-9bfb-bc5e8567c51f.png)

Found a similar issue [here](https://github.com/prescottprue/react-redux-firebase/issues/467) and implemented a fallback for users in the older version of firebase as suggested.

### Check List
- [ ] All tests passing
- [x] Docs updated with any changes or examples if applicable
- [ ] Added tests to ensure new feature(s) work properly

Fixes https://github.com/prescottprue/react-redux-firebase/issues/467

prescottprue commented on src/actions/auth.js:
```diff
@@ -443,21 +443,35 @@ export const login = (dispatch, firebase, credentials) => {
+  let firebaseMethod = firebaseAuth[method]
+
+  // support older versions of firebase that still use signInWithCustomToken & signInWithEmailAndPassword
+  switch (method) {
```
Is it possible for us to fit this logic in the `getLoginMethodAndParams` function?

prescottprue commented on test/utils.js:
```diff
@@ -129,7 +129,7 @@ export const fakeFirebase = {
               providerData: [{}]
             }),
-    signInWithCustomToken: () => {
+    signInAndRetrieveDataWithCustomToken: () => {
```
Nice work updating the tests too!

kadikraman commented:
@prescottprue thanks for the feedback. Updated to move logic to `getLoginMethodAndParams` which incidentally also fixed the tests that were failing before.

I think this definitely needs a test, but I'm not sure how to test for different versions of firebase. Any ideas?

prescottprue commented:
@kadikraman Thanks for making the changes, and nice work!. I can look into good ways to test the different versions.
